app/models/build.rb

```diff
@@ -2,7 +2,7 @@
 class Build < ActiveRecord::Base
   SHA1_REGEX = /\A[0-9a-f]{40}\Z/i
   SHA256_REGEX = /\A(sha256:)?[0-9a-f]{64}\Z/i
-  DIGEST_REGEX = /\A[\w.-]+[\w\/-]*@sha256:[0-9a-f]{64}\Z/i
+  DIGEST_REGEX = /\A[\w.-]+[\w.-]*(:\d+)?[\w\/-]*@sha256:[0-9a-f]{64}\Z/i
```

Comment:
a test for "it allows shas with port" or something like that ?

Reply:
added